chapters/classes.tex

```diff
@@ -383,14 +383,31 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 definition equations is given in \cref{variability-of-expressions}.
 
 \begin{nonnormative}
-A discrete-time variable is a piecewise constant signal which
-changes its values only at event instants during simulation. Such types
-of variables are needed in order that special algorithms, such as the
-algorithm of Pantelides for index reduction, can be applied (it must be
-known that the time derivative of these variables is identical to zero).
-Furthermore, memory requirements can be reduced in the simulation
-environment, if it is known that a component can only change at event
-instants.
+For \lstinline!Real! variables we can distinguish two subtly different categories: discrete-time and piecewise constant, where the discrete-time variables are a subset of all piecewise constant variables.
+The \lstinline!Real! variables declared with the prefix \lstinline!discrete! is a subset of the discrete-time \lstinline!Real! variables.
+For a \lstinline!Real! variable, being discrete-time is equivalent to being assigned in a when-clause.
+A variable used as argument to \lstinline!pre! outside a when-clause must be discrete-time.
+Tools may optimize code to only compute piecewise constant variables at events starting from discrete-time expressions.
```

Comment:
Did you really mean piecewise constant here (instead of discrete-time), sounds like it contradicts the statement below?

Reply:
Yes, will try to clarify.

Reply:
Looks good now.  Resolving.